gems/aws-sdk-ec2/features/step_definitions.rb

```diff
@@ -4,5 +4,24 @@
 end
 
 After("@ec2") do
-  # shared cleanup logic
+end
+
+Given(/^I create a volume$/) do
+  @volume = @service.create_volume(size: 1, availability_zone: 'us-west-2a', volume_type: 'gp2')
+  @volume_id = @volume.id
+end
+
+When(/^I use \#wait_until to wait until volume is available$/) do
+  expect {
+    @resp = @volume.wait_until {|v| v.state == 'available'}
+  }.not_to raise_error
+end
+
+Then(/^Waiter works and volume is cleaned up$/) do
```

Comment:
Use the cleanup logic you stubbed above. If the test fails you won't clean up the resource as written now.

Reply:
Make sense, fixed in the latest commit.